### PR TITLE
net: wifi_mgmt: Add support for power save timeout configuration

### PR DESCRIPTION
**Pull request by ajayparida**
Add support for configuring power save timeout in Wi-Fi chipsets. Changes to configure power save inactivity timer.

Signed-off-by: Ajay Parida <ajay.parida@nordicsemi.no>

### CODE REVIEW

**Comment by rlubos** [comment]
The commit title should be prefixed with `[nrf fromtree] ` tag. Also, please cherry-pick with `-x` option, so that upstream commit is referenced.

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
Looks good, but as with other `sdk-zephyr` cherry-picks, it needs an ACK from the release team.

File: subsys/net/l2/wifi/wifi_shell.c
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:883]: Ideally the API can also be called set_ps_inactivitity, but I guess this is already merged upstream, so, lets leave it for now.

**Review by krish2718 — CHANGES_REQUESTED** [review]
The cherry-pick commit in commit log looks wrong, it has to be upstream commit.

**Comment by krish2718** [comment]
> The cherry-pick commit in commit log looks wrong, it has to be upstream commit.

Looks correct, sorry.

**Review by krish2718 — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**
